scripts/nbd_client_manager.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/python
+
+"""
+Provides functions and a CLI for safely connecting to and disconnecting from
+NBD devices.
+"""
+
+import argparse
+import logging
+import os
+import subprocess
+import time
+import fcntl
+
+
+LOGGER = logging.getLogger("nbd_client_manager")
+LOCK_FILE = '/var/run/nonpersistent/nbd_client_manager'
+
+
+class NbdDeviceNotFound(Exception):
+    """
+    The NBD device file does not exist. Raised when there are no free NBD
+    devices.
+    """
+    def __init__(self, nbd_device):
+        self.nbd_device = nbd_device
+
+
+class FileLock(object):
+    """Container for data relating to a file lock"""
+    def __init__(self, path):
+        self._path = path
+        self._lock_file = None
+
+    def lock(self):
+        """Acquire the lock"""
+        flags = fcntl.LOCK_EX
+        self._lock_file = open(self._path, 'w+')
+        fcntl.flock(self._lock_file, flags)
+
+    def unlock(self):
```

Comment:
if you add `__enter__` and `__exit__` as alias of these you can use `with FILE_LOCK: ` below

Reply:
Thanks, that's a great idea, I'll add those :)